connect/runtime/src/main/java/org/apache/kafka/connect/storage/OffsetUtils.java

```diff
@@ -48,8 +48,8 @@ public static void validateFormat(Object offsetData) {
             if (value == null)
                 continue;
             Schema.Type schemaType = ConnectSchema.schemaType(value.getClass());
-            if (!schemaType.isPrimitive())
-                throw new DataException("Offsets may only contain primitive types as values, but field " + entry.getKey() + " contains " + schemaType);
+            if (schemaType == null || !schemaType.isPrimitive())
+                throw new DataException("Offsets may only contain primitive types as values, but field " + entry.getKey() + " contains " + value.getClass());
```

Comment:
`schemaType` is null means that the value type is not supported by the Connect's Data API. May be we should throw an exception with a message indicating this.